doc/source/contribute.rst

```diff
@@ -6,16 +6,53 @@ Contribute
 How to Contribute to this Project
 ---------------------------------
 
-Do you know or are willing to learn C# and the **KSP** public API? Great, we could use your help! The source code for **kOS** is kept on `github`_ and is currently maintained by the `kOS Team`_. The first steps you will want to take is to clone the latest version of **kOS** from github and try to build it using your favorite C# compiler suite.
+Do you know or are willing to learn C# and the **KSP** public API? Great, we could use your help! The source code for **kOS** is kept on `github`_ under http://github.io/KSP-KOS/KOS.
+
+If you are already quite familiar with git and Github, the usual Github project development path is used:
+
+  - Tell github to fork the main repository to your own github clone of it.
+  - Clone your fork to your local computer.
+  - On your local computer, make a branch from ``develop`` (don't edit ``develop`` directly) and make your changes in your branch.
+  - Commit your changes and push them up to the same branch name on your github fork.
+  - Make a Pull Request on Github to merge the branch from your fork to the ``develop`` branch of the main repository.
+  - Wait for a developer to notice the Pull Request and start examining it.  There should be at the very least a comment letting you know it's being looked at, within a short time.  KSP-KOS is quite actively developed and someone should notice it soon.
+
+  - Your request is more likely to get merged quickly if you make sure the ``develop`` branch you start from is always up to date with the latest upstream develop when you first split your branch from it.  If it takes a long time to finish, it may be a good idea to check again before making the Pull Request to see if there's been any new upstream ``develop`` changes, and merge them into your branch yourself so the rest of the team has an easier time deciphering the git diff output.
```

Comment:
Should this also comment that we accept both upstream merges and rebases, but that we ask you don't rebase if the pull request is being relied on for changes in other pull requests?